spring-cloud-azure-appconfiguration-config/src/main/java/com/microsoft/azure/spring/cloud/config/AzureConfigPropertySource.java

```
@@ -135,15 +147,12 @@ public FeatureSet initProperties(FeatureSet featureSet) throws IOException {
     private String getKeyVaultEntry(String value) {
         String secretValue = null;
         try {
-            String stringUri = "";
             URI uri = null;
-            Secret secret = new Secret();
 
             // Parsing Key Vault Reference for URI
             try {
                 JsonNode kvReference = mapper.readTree(value);
-                stringUri = kvReference.at("/uri").asText();
-                uri = new URI(stringUri);
+                uri = new URI(kvReference.at("/uri").asText());
             } catch (URISyntaxException e) {
                 if (azureProperties.isFailFast()) {
                     LOGGER.error("Error Processing Key Vault Entry URI.", e);
```

Seems should throw exception in the fail fast condition?